src/jit/compiler.hpp

```diff
@@ -3020,6 +3020,35 @@ inline unsigned Compiler::fgThrowHlpBlkStkLevel(BasicBlock* block)
 */
 inline void Compiler::fgConvertBBToThrowBB(BasicBlock* block)
 {
+    // If we're converting a BBJ_CALLFINALLY block to a BBJ_THROW block,
+    // then mark the subsequent BBJ_ALWAYS block as unreferenced.
+    if (block->isBBCallAlwaysPair())
+    {
+        BasicBlock* leaveBlk = block->bbNext;
+        noway_assert(leaveBlk->bbJumpKind == BBJ_ALWAYS);
+
+        leaveBlk->bbFlags &= ~BBF_DONT_REMOVE;
+        leaveBlk->bbRefs = 0;
+        leaveBlk->bbPreds = nullptr;
+
+#if FEATURE_EH_FUNCLETS && defined(_TARGET_ARM_)
+        // This function (fgConvertBBToThrowBB) can be called before the predecessor lists are created (e.g., in fgMorph).
+        // The fgClearFinallyTargetBit() function to update the BBF_FINALLY_TARGET bit depends on these predecessor lists.
+        // If there are no predecessor lists, we immediately clear all BBF_FINALLY_TARGET bits (to allow subsequent dead
+        // code elimination to delete such blocks without asserts), and set a flag to recompute them later, before they are
+        // required.
+        if (fgComputePredsDone)
+        {
+            fgClearFinallyTargetBit(leaveBlk->bbJumpDest);
+        }
+        else
+        {
+            fgClearAllFinallyTargetBits();
```

Comment:
You still could end up clearing the bits multiple times. I suppose that won't happen often enough to worry about.

So I'm ok with what you have.